### PR TITLE
AB#86890 incorrect error order draft

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using HE.Investment.AHP.Domain.FinancialDetails.Constants;
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs:1]: View coupling to Domain?
No way.

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs:1]: It's the same as on other sections and only this one bothers you? 
image

Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs:1]: Question is what with below examples of coupling:
image
image
image
We want to decouple view form domain at all cost?

Comment (Antek012) [reply src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs]: I believes it should be in Contract so both Domain and View can use.

File: src/HE.Investments.Common.WWW/Extensions/ModelStateExtensions.cs [hunk 2]
```diff
@@ -22,25 +22,25 @@ public static (bool HasAnyError, string Messge) GetErrors(this ModelStateDiction
...
             var (hasError, errorMsg) = GetErrors(modelState, key);
+
             if (hasError)
             {
                 result.TryAdd(key, errorMsg);
             }
         }
 
-        return result;
+        return result.Count > 0 ? result : null;
```
Comment (Antek012) [inline src/HE.Investments.Common.WWW/Extensions/ModelStateExtensions.cs:43]: is better to return empty collection to avoid null handling in callers

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investments.Common.WWW/Extensions/ModelStateExtensions.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investments.Common.WWW/Extensions/ModelStateExtensions.cs]: agree, removed null return part

File: src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/FinancialDetails/Consts/FinancialDetailsValidationFieldsOrder.cs]: some of the constants were moved to conttract

**Pull request merged**